sqlmesh/core/context.py

```diff
@@ -1972,6 +1972,25 @@ def print_info(self, skip_connection: bool = False, verbose: bool = False) -> No
         if state_connection:
             self._try_connection("state backend", state_connection.connection_validator())
 
+    @python_api_analytics
+    def print_environment_names(self, show_expiry: bool) -> None:
+        """Prints all environment names along with expiry datetime if show_expiry is True."""
+        environment_names = self._new_state_sync().get_environment_names(get_expiry_ts=show_expiry)
+        if not environment_names:
+            raise SQLMeshError(
+                "This project has no environments. Create an environment using the `sqlmesh plan` command."
+            )
+        output = (
```

Comment:
Can we have a method in `Console` called `print_environments` and do all the formatting there instead? This way we can override what the output looks like depending on which environment we're in.